infinispan-remote/programmatic-protostream-marshalling/src/main/java/org/infinispan/tutorial/simple/remote/marshalling/InfinispanProgrammaticProtostreamMarshalling.java

```diff
@@ -0,0 +1,91 @@
+package org.infinispan.tutorial.simple.remote.marshalling;
+
+import static org.infinispan.query.remote.client.ProtobufMetadataManagerConstants.PROTOBUF_METADATA_CACHE_NAME;
+import static org.infinispan.tutorial.simple.connect.TutorialsConnectorHelper.TUTORIAL_CACHE_NAME;
+
+import java.time.YearMonth;
+import java.util.Arrays;
+import java.util.Collections;
+
+import org.infinispan.client.hotrod.RemoteCache;
+import org.infinispan.client.hotrod.RemoteCacheManager;
+import org.infinispan.client.hotrod.configuration.ConfigurationBuilder;
+import org.infinispan.commons.marshall.ProtoStreamMarshaller;
+import org.infinispan.protostream.FileDescriptorSource;
+import org.infinispan.protostream.SerializationContext;
+import org.infinispan.protostream.schema.Schema;
+import org.infinispan.tutorial.simple.connect.TutorialsConnectorHelper;
+
+/**
+ * Infinispan Server includes a default property realm that requires
+ * authentication. Create some credentials before you run this tutorial.
+ *
+ */
+public class InfinispanProgrammaticProtostreamMarshalling {
+
+   static Schema schema = MagazineSchemaCreator.magazineSchema();;
+   static RemoteCacheManager client;
+   static RemoteCache<String, Magazine> magazineRemoteCache;
+
+   public static void main(String[] args) throws Exception {
+      connectToInfinispan();
+
+      manipulateCache();
+
+      disconnect(false);
+   }
+
+   static void connectToInfinispan() {
+      ConfigurationBuilder builder = TutorialsConnectorHelper.connectionConfig();
+
+      ProtoStreamMarshaller marshaller = new ProtoStreamMarshaller();
+      SerializationContext serializationContext = marshaller.getSerializationContext();
+      FileDescriptorSource fds = FileDescriptorSource.fromString(schema.getName(), schema.toString());
+      serializationContext.registerProtoFiles(fds);
+      serializationContext.registerMarshaller(new MagazineMarshaller());
+      builder.marshaller(marshaller);
+      builder.remoteCache(TUTORIAL_CACHE_NAME).marshaller(marshaller);
+
+      // Connect to the server
+      client = TutorialsConnectorHelper.connect(builder);
+
+      // Create and add the Protobuf schema in the server
+      registerMagazineSchemaInTheServer(client);
```

Comment:
This is optional for this tutorial and can be removed.

Reply:
if I go to the console, I can't display the data. for me, this is not optional. it adds confusion and understanding